sunpy/net/helioviewer.py

```diff
@@ -132,11 +132,17 @@ def get_closest_image(self, date, sourceid):
 
         return response
 
-    def download_jp2(self, date, sourceid, directory=None, overwrite=False, **kwargs):
+    def display_source_id(self, observatory, instrument, detector, measurement):
+        """
+        Returns a dictionary containing the source ID of all the 
+        images in the data source
+        """
+        return self._source_dict[(observatory, instrument, detector, measurement)]
```

Comment:
I don't see why we need this function.

Reply:
I just made two functions. One for downloading the image using the old API and the other which is hidden from the user which actually uses the new API to download the image. We call the function download_jp2_ implicitly from download_jp2 giving user the vision that they are using older API but in fact the code uses the new API

Reply:
But I don't understand why it has to be two functions? I think you only need one.

Reply:
We need two functions to make sure that source ID is passed to the helioviewer API. If we look at the download_jp2_ function all it do is passes the parameters namely date, source ID to the API and it take these values from download_jp2 function. We have to make a function where source ID is a required parameter because it is compulsory for us to pass it. That is why I made two functions one which is hidden from the user and one which the user interacts with i.e download_jp2

Reply:
But you don't need to pass in sourceid into the `download_jp2`, you can work it out and pass it inside `download_jp2_`.

Reply:
I am not passing the source ID in download_jp2. For the user, we are working in the same old way. I am passing the source ID implicitly to download_jp2_ using download_jp2 function

Reply:
I just passed the source ID to download_jp2_ function from download_jp2 function, but this won't be visible to the user using the API

Reply:
But it doesn't need two functions. You can merge them into one.

Reply:
We cannot write this " "sourceId" : sourceid " in the params until we have source ID as a parameter
I have tried and it gives an error. 

What is the issue if we create two funtions?

Reply:
Nothing is wrong with two functions if we need to do it.

I just don't think we need to.

Reply:
I tried doing it with one function when I started but I was not able to implement it.  That is why I implemented it using two functions. I am still not very clear how can we implement the same using just one function

Reply:
Maybe we are talking over each other but `"sourceId" : sourceid` you can work out the sourceid the line above the params section without another function for downloading.

Reply:
Yes we can. But then if we write "sourceID" : sourceid in params it throws an error saying that sourceID is not defined

Reply:
Sure, but above we define sourceid above by accessing the dictionary.

Reply:
Ok I will give it a try to implement it using just one function